mmr/mmr.go

```diff
@@ -434,9 +437,13 @@ func (m *Proof) CalculateRootWithNewLeaf(leaves []types.Leaf, newIndex uint64, n
 			i++
 		}
 
-		reversePeahashOfKes := peaksHashes[i:]
-		reverse(reversePeahashOfKes)
-		peaksHashes = append(peaksHashes[:i], reversePeahashOfKes...)
+		var reversedHashes [][]byte
```

Comment:
given that you know the length of this slice, wouldn't it make sense to pre-allocate all the memory upfront?

Reply:
pre-allocating the memory upfront seemed to take more processing time. Here's result from a profile while running the benchmark, pre-allocating the slice. Line 34 is where the pre-allocation was done, and it was taking 3.07 seconds for a slice of length 100000. 

```
(pprof) list GetElem 
Total: 290.18s
ROUTINE ======================== github.com/ComposableFi/go-merkle-trees/mmr.(*Batch).GetElem in /Users/klinsmann/go/src/github.com/ComposableFi/merkle-go/mmr/mmr_store.go
         0     32.16s (flat, cum) 11.08% of Total
         .          .     29:func (b *Batch) append(pos uint64, elems [][]byte) {
         .          .     30:   b.memoryBatch = append(b.memoryBatch, BatchElem{pos, elems})
         .          .     31:}
         .          .     32:
         .          .     33:func (b *Batch) GetElem(pos uint64) []byte {
         .      3.07s     34:   memoryBatch := make([]BatchElem, len(b.memoryBatch))
         .      4.57s     35:   copy(memoryBatch, b.memoryBatch)
         .     24.52s     36:   reverse(memoryBatch)
         .          .     37:
         .          .     38:   for _, mb := range memoryBatch {
         .          .     39:           startPos, elems := mb.pos, mb.elems
         .          .     40:           if pos < startPos {
         .          .     41:                   continue

```

Reply:
Initialization shouldn't be part of the benchmark right? We're only interested in the get part given that the tree is big